libs/core/langchain_core/utils/function_calling.py

```diff
@@ -188,15 +250,38 @@ def convert_to_openai_function(
         A dict version of the passed in function which is compatible with the
             OpenAI function-calling API.
     """
+    from langchain_core.tools import BaseTool
+
     if isinstance(function, dict):
         return function
     elif isinstance(function, type) and issubclass(function, BaseModel):
         return cast(Dict, convert_pydantic_to_openai_function(function))
+    elif isinstance(function, BaseTool):
+        return format_tool_to_openai_function(function)
     elif callable(function):
         return convert_python_function_to_openai_function(function)
-
     else:
         raise ValueError(
             f"Unsupported function type {type(function)}. Functions must be passed in"
             f" as Dict, pydantic.BaseModel, or Callable."
         )
+
+
+def convert_to_openai_tool(
```

Comment:
Should we add a 'Runnable' support while we're at it?

Reply:
how would we convert it?

Reply:
using input_schema and docstrings?